lib/std/Thread/RwLock.zig

```diff
@@ -246,4 +247,132 @@ pub const DefaultRwLock = struct {
         if ((state & READER_MASK == READER) and (state & IS_WRITING != 0))
             rwl.semaphore.post();
     }
+
+    /// The instantaneous value of the waiting-to-write state, for testing only.
+    fn isWaitingToWrite(rwl: *DefaultRwLock) bool {
+        const state = @atomicLoad(usize, &rwl.state, .SeqCst);
+        return (state & DefaultRwLock.WRITER_MASK) != 0;
+    }
```

Comment:
1. having no writers could still mean that there's the one that decremented its WRITER count while adding the IS_WRITING bit
2. This is implementation specific and probably shouldn't be tested on since the implementation could change with there being no way to know or observe pending writers (i.e. TID based locking or queue based locking).

Reply:
There is no other way I know of to test that the WRITER flag is being subtracted and the bug is fixed. It is used only in an implementation specific test.

Reply:
Maybe something like this:
```zig
rwlock = RwLock{}
rwlock.lock();
rwlock.unlock();
expectEqual(rwlock, RwLock{})
```

Reply:
Sure, I will do that change, although comparing internal state for equality is also somewhat implementation specific, so I should probably only do it for DefaultRwLock.

I very surprised that white box testing is not used -- is this the Zig convention, or your personal convention?

Reply:
Yes it should only be done for DefaultRwLock if the plan is to keep the implementation specific check. 

Im not sure about testing methodologies; The suggestion was just a way to test for the leaking WRITER increment in a way that looked a bit simpler. For sync primitives in particular, I just write tests (often in the form of benchmarks) which either increases the chances of various interleavings, or ensures deadlock/starvation properties are accounted for.